WooCommerce/src/main/kotlin/com/woocommerce/android/ui/analytics/ranges/AnalyticsHubDateRangeSelection.kt

```diff
@@ -22,19 +25,51 @@ import com.woocommerce.android.ui.analytics.ranges.data.YearToDateRangeData
 import com.woocommerce.android.ui.analytics.ranges.data.YesterdayRangeData
 import java.util.Calendar
 import java.util.Date
+import java.util.Locale
 
-class AnalyticsHubDateRangeSelection(
-    selectionType: SelectionType,
-    currentDate: Date = Date(),
-    calendar: Calendar = Calendar.getInstance()
-) {
-    val currentRange: AnalyticsHubTimeRange?
-    val previousRange: AnalyticsHubTimeRange?
+/**
+ * This class represents the date range selection for the Analytics Hub
+ *
+ * You can create it through the [AnalyticsHubDateRangeSelection.SelectionType.generateSelectionData]
+ * function since it will return the correct data for the given selection type
+ *
+ * When creating the object through the available constructor, the Selection will be set as [CUSTOM]
+ */
+class AnalyticsHubDateRangeSelection {
+    val selectionType: SelectionType
+    val currentRange: AnalyticsHubTimeRange
+    val previousRange: AnalyticsHubTimeRange
+    val currentRangeDescription: String
+    val previousRangeDescription: String
 
-    init {
-        val rangeData = generateTimeRangeData(selectionType, currentDate, calendar)
-        this.currentRange = rangeData.currentRange
-        this.previousRange = rangeData.previousRange
+    constructor(
```

Comment:
A neat idea with two constructors 👏  . Can we make this `private` as well, please? It'll be easier to read and user correctly this way:
```suggestion
    private constructor(
```

Reply:
Sure! Fixed here: https://github.com/woocommerce/woocommerce-android/pull/8113/commits/7cd94ffb8aa5f83c121ecb0cd6bfaebebc6802df